tests/macos_frames.rs

```diff
@@ -0,0 +1,33 @@
+// Copied from https://github.com/rust-lang/rust/blob/2cb0b8582ebbf9784db9cec06fff517badbf4553/src/test/ui/issues/issue-45731.rs
+// This needs to go in its own file, since it modifies the dSYM
+// for the entire test
+#[test]
+#[cfg(target_os = "macos")]
+fn simple_test() {
+    use std::{env, fs, panic};
+
+    // Find our dSYM and replace the DWARF binary with an empty file
+    let mut dsym_path = env::current_exe().unwrap();
+    let executable_name = dsym_path.file_name().unwrap().to_str().unwrap().to_string();
+    assert!(dsym_path.pop()); // Pop executable
+    dsym_path.push(format!(
+        "{}.dSYM/Contents/Resources/DWARF/{0}",
+        executable_name
+    ));
+    {
+        let _ = fs::OpenOptions::new()
+            .read(false)
+            .write(true)
+            .truncate(true)
+            .create(false)
+            .open(&dsym_path)
+            .unwrap();
+    }
+
+    env::set_var("RUST_BACKTRACE", "1");
+
+    // We don't need to die of panic, just trigger a backtrace
+    let _ = panic::catch_unwind(|| {
+        assert!(false);
+    });
```

Comment:
I don't believe that this is testing what it's meant to test, you probably want to avoid panicking/env vars here and instead call `Backtrace::new()`